File: src/test/java/CheckDropdown.java
import com.microsoft.playwright.*;

import java.nio.file.Files;
import java.nio.file.Paths;

public class CheckDropdown {
    public static void main(String[] args) {
        try (Playwright playwright = Playwright.create()) {
            Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setHeadless(true));
            Page page = browser.newPage();
            page.navigate("https://casekaro.com/");

            Locator searchIcon = page.locator("summary.header__icon--search").first();
            if (searchIcon.isVisible()) searchIcon.click();

            Locator searchInput = page.locator("input[placeholder*='search']").first();
            if (!searchInput.isVisible()) {
                searchInput = page.locator("input[type='search']").first();
            }
            searchInput.fill("iPhone 16 Pro");

            page.waitForTimeout(4000);
            
            try {
                String html = page.content();
                Files.write(Paths.get("debug_page.html"), html.getBytes());
                System.out.println("✅ Saved debug_page.html");
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }
}

File: check_dropdown.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.goto('https://casekaro.com/');
  
  await page.locator('input[placeholder*="search"]').first().fill('iPhone 16 Pro');
  
  await page.waitForSelector('.snize-ac-results', { state: 'visible', timeout: 10000 });
  await page.waitForTimeout(2000); // wait for results to populate
  
  const suggestions = await page.$$('.snize-suggestion, .snize-ac-results li, .snize-ac-results a, .snize-title');
  for (const s of suggestions) {
    const text = await s.innerText();
    const className = await s.getAttribute('class');
    const href = await s.getAttribute('href');
    console.log(`Class: ${className} | Text: '${text}' | Href: ${href}`);
  }
  
  await browser.close();
})();
